index.js

```diff
@@ -119,16 +100,81 @@
   //  $$type :: String
   var $$type = '@@type';
 
-  //  type :: Any -> String
+  //  RPARSE :: RegExp
+  var RPARSE = /^([^]+)\/([^]+?)(?:@(\d+))?$/;
+
+  //       TypeIdentifier :: (Nullable String, String, Number)
+  //                      -> TypeIdentifier
+  function TypeIdentifier(namespace, name, version) {
+    return {
+      namespace: namespace,
+      name: name,
+      version: version
+    };
+  }
+
+  //       parseTypeIdentifier :: String -> TypeIdentifier
+  function parseTypeIdentifier(s) {
+    var parsed = RPARSE.exec(s);
+    if (parsed == null) return TypeIdentifier(null, s, 0);
+    return TypeIdentifier(
+      parsed[1] == null ? null : parsed[1],
+      parsed[2],
+      parsed[3] == null ? 0 : Number(parsed[3])
+    );
+  }
+
+  //       identifyNativeType :: Any -> TypeIdentifier
+  function identifyNativeType(x) {
+    return TypeIdentifier(
+      null,
+      Object.prototype.toString.call(x).slice('[object '.length, -']'.length),
+      0
+    );
+  }
+
+  //. ### Usage
+  //.
+  //. #### With native types
+  //.
+  //. ```javascript
+  //. > type(null);
+  //. {namespace: null, name: 'Null', version: 0}
+  //.
+  //. > type(true);
+  //. {namespace: null, name: 'Boolean', version: 0}
+  //.
+  //. > type([1, 2, 3]);
+  //. {namespace: null, name: 'Array', version: 0}
+  //. ```
+  //.
+  //. #### With custom types
+  //.
+  //. ```javascript
+  //. > function Identity(x) { this.value = x; }
+  //. > Identity['@@type'] = 'my-package/Identity@1';
+  //. > type(Identity);
+  //. {namespace: null, name: 'Function', version: 0}
```

Comment:
before, returning `Function` here made sense. as if one had reference to some type representative it could just see what `@@type` contained. but now as the `@@type` needs to be parsed before it's usable one needs to first create some value of the type, so maybe we should export `parseTypeIdentifier` ?